### ISSUE TITLE
[Bug]: Visual modeling- Linear Modeling failed in 0.16.4 

### ISSUE THREAD

**Issue by ImpBIS** [jasp-stats/jasp-issues#1849]
### JASP Version

0.16.4

### JASP Module

jaspVisualModeling

### What analysis are you seeing the problem on?

Linear Modeling, generalised linear modeling

### What OS are you seeing the problem on?

Win 10

### Bug Description

I attempted to follow the visual modeling example found here https://www.youtube.com/watch?v=YhrgsHuxGBc
The dataset is available here (see link on video page)  https://www.youtube.com/redirect?event=video_description&redir_token=QUFFLUhqbW1wN0c4am1RMnNrdEJHUzNpTDBDVDdTN0ZpZ3xBQ3Jtc0ttUGFDWWpuei1WQ282RVk3YWZCSFUtRjZmb2FwUFl3d0VsRHEwOXRQY19ISFZqdEtUUzhSSnZNM2Y2aTljdU1xQ0lFYkxCR09TanFEZ2k1bWFvcFVzRURRRHdxRFByNENsRDNDemRGeVpJTHA3YWpoUQ&q=https%3A%2F%2Fjournals.plos.org%2Fplosone%2Farticle%3Fid%3D10.1371%2Fjournal.pone.0220784%23sec017&v=YhrgsHuxGBc 

Flexplot in the module worked OK

The linear plot example (at around 3.50 min) in the video failed at the first step entering the dependent variable with error message
![image](https://user-images.githubusercontent.com/115127151/194200857-b15a9d4c-8a16-480b-a40e-f44cd160655d.png)

Similarly for general linear modeling an error message occurred when entering the independent variable
![image](https://user-images.githubusercontent.com/115127151/194201108-58e1eb5b-a69c-4a9e-8f15-d6145d802c04.png)

### Expected Behaviour

On another PC that still had JASP version 0.16.3 the module worked perfectly and reproduced the analysis shown in the video.
<img width="945" alt="2022-10-06_125036" src="https://user-images.githubusercontent.com/115127151/194204027-a622a753-c2b4-4032-b0e1-095e66f75e4a.png">

### Steps to Reproduce

I tried the analysis with JASP 0.16.3 and it worked albeit on a different PC but same Win version
On JASP 0.16.3
<img width="945" alt="2022-10-06_125036" src="https://user-images.githubusercontent.com/115127151/194203442-24696a40-b938-4667-a889-731877d8b3f5.png">
On JASP 0.16.4
![image](https://user-images.githubusercontent.com/115127151/194203519-6f8e6254-fd48-4ab4-9cea-d0709e3552e8.png)

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if any.
- [X] I have included a JASP file that causes the crash/bug, if any.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by boutinb** [comment]
Hi @ImpBIS, I can reproduce your problem with this data:
[pone.0220784.s001.csv](https://github.com/jasp-stats/jasp-issues/files/9722901/pone.0220784.s001.csv)
But I can reproduce the same problem in 0.16.3. Can you try it again in 0.16.3 with this data?

@dustinfife Could you have.a look also? If you load this file, start Linear Modeling, and set CESD as Dependent Variable then you get the error:
```
Error in `$<-.data.frame`(`*tmp*`, 'model', value = structure(list(coefficients = c(`(Intercept)` = 16.2266401590457), : replacement has 11 rows, data has 1
```

**Comment by juliuspf** [comment]
@dustinfife

**Comment by dustinfife** [comment]
There were two problems that were fixed with [this commit](https://github.com/dustinfife/flexplot/commit/43b77ea4627d683dfb8bc862a98a93fcaf8d75ef). Now there's another problem. With the default jasp theme, I get an error that says, "Error in getGraphOption('bty'): could not find function 'getGraphOption'". 

Did something change on JASP's end with that function?

**Comment by ImpBIS** [comment]
I retried the pone.0220784.s001.csv on 0.16.4. I could not get the linear modeling visualisations to work at all.

I then tried this on 0.16.3. If I load the dependent variable first the error message occurs. Continuing to load the independent variables causes the error message to disappear and the visualisations work. This seems to be the case irrespective of the variable chosen.

I then tried a completely different dataset of my own and got the same result: will not run in 0.16.4 and in 0.16.3 runs if independent variables chosen first, then dependent variable. Or if choosing dependent first  the error message occurs but is "overcome" when the independent variables are subsequently chosen.

**Comment by dustinfife** [comment]
It won't work yet. I've made changes on my end, then they have to be integrated in JASP's app.

**Comment by ImpBIS** [comment]
Many thanks.

**Comment by vandenman** [comment]
@dustinfife did you call `jaspGraphs::getGraphOption` or just `getGraphOption`? I suspect we may have removed `library(jaspGraphs)` somewhere. Explicitly prefixing with `jaspGraphs::` should solve that.

**Comment by dustinfife** [comment]
[This commit ](https://github.com/dustinfife/flexplot/commit/5e76702062347bf9fdcdacd5ac4a9bef83cd1d1a)fixes that. Thanks @vandenman.

**Comment by juliuspf** [comment]
@dustinfife, can you answer this question: https://github.com/jasp-stats/jasp-issues/issues/1895#issuecomment-1309041565

**Comment by dustinfife** [comment]
This has been fixed on my end. I think the nightlies implement this fix as well (is that right @boutinb ?)

**Comment by boutinb** [comment]
> This has been fixed on my end. I think the nightlies implement this fix as well (is that right @boutinb ?)

The Visual Modeling module should take automatically the last version of the flexplot package. So it should be available in the Nightly version: https://static.jasp-stats.org/Nightlies/ . WARNING: This version is in development! But should be good enough to check whether the issue is fixed.
Reopen this issue, if it is not.

**Comment by dustinfife** [comment]
I learned this was also a problem for the generalized linear model module. [This commit ](https://github.com/dustinfife/flexplot/commit/43b77ea4627d683dfb8bc862a98a93fcaf8d75ef)fixes it.

**Comment by boutinb** [comment]
> I learned this was also a problem for the generalized linear model module. [This commit ](https://github.com/dustinfife/flexplot/commit/43b77ea4627d683dfb8bc862a98a93fcaf8d75ef)fixes it.

This commit was made in October, so the last nightlies should have incorporated this commit.

**Comment by dustinfife** [comment]
Sorry, I linked the wrong commit. [Here is the right one.](https://github.com/dustinfife/flexplot/commit/0d5b6a88df59f82cb0dc77e81ba9416113260a83)

**Comment by boutinb** [comment]
Hi @dustinfife I see your commits are in the development branch. I think they have to be merged into the master branch.

**Comment by dustinfife** [comment]
They're merged now.

**Comment by tomtomme** [comment]
Just tested the updated module on linux (bioarch repo version) and it mostly works now.

switching to "quadratic" for the fitted line produces
![grafik](https://user-images.githubusercontent.com/4428185/203288163-12f99670-1a9b-43b3-b759-b3b2450ea703.png)

this was not repdroducable however. It seem to need a combination of clicked checkboxes or a special order of when to click what. 

A reproducable error was with clicking on "add as polynomial" in "model terms"
![grafik](https://user-images.githubusercontent.com/4428185/203289454-28c54452-d9ee-42b5-9e4e-ed975172763b.png)


I tested with the beer googles dataset from the data-library.
I switched attractiveness to scale to be able to use it for the module.
